src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationILGen.cs

```diff
@@ -65,6 +65,15 @@ internal ModuleBuilder ModuleBuilder
         }
         internal TypeAttributes TypeAttributes { get { return _typeAttributes; } }
 
+        private static readonly string[] s_typeString = new string[] { "type" };
+        private static readonly string[] s_xmlReaderString = new string[] { "xmlReader" };
+        private static readonly string[] s_objectToSerializeWriterString = new string[] { "objectToSerialize", "writer" };
+        private static readonly string[] s_readerString = new string[] { "reader" };
+        private static readonly Type[] s_typeType = new Type[] { typeof(Type) };
+        private static readonly Type[] s_xmlReaderType = new Type[] { typeof(XmlReader) };
+        private static readonly Type[] s_objectToSerializeWriterType = new Type[] { typeof(object), typeof(XmlSerializationWriter) };
+        private static readonly Type[] s_readerType = new Type[] { typeof(XmlSerializationReader) };
```

Comment:
```suggestion
        private static readonly Type[] s_xmlReaderType = new Type[] { typeof(XmlSerializationReader) };
```
given  you have readerString distinct from xmlReaderString